standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2104,9 +2106,14 @@ public void truncateTable(String catName, String dbName, String tableName, List<
   private void truncateTableInternal(String catName, String dbName, String tableName,
       List<String> partNames, String validWriteIds, long writeId, boolean deleteData)
           throws TException {
+    EnvironmentContext context = new EnvironmentContext();
+    if (tableName.contains(".")) {
```

Comment:
wait, are we doing this because we concatenate in API? can we extend API with `String tableMetaRef`

Reply:
As discussed offline, resloving